manual_hook.rb

```diff
@@ -2,56 +2,75 @@
 
 require 'resolv'
 
-def setup_dns(domain, txt_challenge)
-  resolved = false;
-  singleLoop = false;
-  dns = Resolv::DNS.new;
-  acme_domain = "_acme-challenge."+domain; 
-  puts "Checking for pre-existing TXT record for the domain: \'#{acme_domain}\'."
-
-  until resolved
-    dns.each_resource(acme_domain, Resolv::DNS::Resource::IN::TXT) { |resp|
-     if resp.strings[0] == txt_challenge
-       puts "Found #{resp.strings[0]}. match."
-       resolved = true;
-     else
-       puts "Found #{resp.strings[0]}. no match."
-     end
-    }
-
-    if !resolved
-     if !singleLoop
-       puts "Create TXT record for the domain: \'#{acme_domain}\'. TXT record:"
-       puts "\'#{txt_challenge}\'"
-       puts "Press enter when DNS has been updated..."
-       $stdin.readline()
-       singleLoop = true
-     end
-
-     puts "Didn't find a match for #{txt_challenge}"; 
-     puts "Waiting to retry..."; 
-     sleep 30; 
+# Struct to store a challenge
+Challenge = Struct.new(:domain, :acme_domain, :txt_challenge)
+
+
+# Check if a challenge is resolved, returns true in that case
+def resolved?(dns, challenge)
+  dns.each_resource(challenge[:acme_domain], Resolv::DNS::Resource::IN::TXT) { |resp|
+    if resp.strings[0] == challenge[:txt_challenge]
+      puts "Found #{resp.strings[0]}. match."
+      return true
+    else
+      puts "Found #{resp.strings[0]}. no match."
+      return false
+    end
+  }
+  puts "Didn't find a match for #{challenge[txt_challenge]}"
+  return false
+end
+
+def setup_dns(challenges)
+  # DNS Resolver
+  dns = Resolv::DNS.new
+  first_iteration = true
+  until challenges.empty? # Until all challenges are resolved
+    challenges.delete_if do |challenge| # Delete resolved challenges
+      puts "Checking for TXT record for the domain: \'#{challenge[:acme_domain]}\'."
+      resolved?(dns, challenge)
+    end
+
+    if first_iteration
+      challenges.each do |challenge|
+        puts "Create TXT record for the domain: \'#{challenge[:acme_domain]}\'. TXT record:"
+        puts "\'#{challenge[:txt_challenge]}\'"
+      end
+      puts "Press enter when DNS has been updated..."
+      $stdin.readline
+      first_iteration = false
+    else
+      unless challenges.empty?
+        puts "Waiting to retry..."
+        sleep 30
+      end
     end
   end
 end
 
-def delete_dns(domain, txt_challenge)
+def delete_dns(challenges)
   puts "Challenge complete. Leave TXT record in place to allow easier future refreshes."
 end
 
 if __FILE__ == $0
-  hook_stage = ARGV[0]
-  domain = ARGV[1]
-  txt_challenge = ARGV[3]
+  # puts "ARGV: #{ARGV.inspect}"
+  hook_stage = ARGV.shift
+  challenges = []
+  while ARGV.length >= 3
+    domain = ARGV.shift
+    acme_domain = "_acme-challenge.#{domain}"
```

Comment:
`acme_domain = "_acme-challenge.#{domain}".sub(/\.\*\./, ".")` via @rudiedirkx patch in https://github.com/jbjonesjr/letsencrypt-manual-hook/issues/10#issuecomment-361069448.

wildcard certificate support: https://community.letsencrypt.org/t/acme-v2-and-wildcard-certificate-support-is-live/55579

What's interesting is that the acme challenge domain drops the asterisk completely to just create a top level certificate. I'll have to give that some 👀